### PR TITLE
0.3.4 final

### PR DESCRIPTION
**Pull request by AionJayT**
## Notice

It is not allowed to submit your PR to the master branch directly, please submit your PR to the master-pre-merge branch.

## Description

It is the PR merge for all the 0.3.4 features to get  into master

Fixes Issue # .

## Type of change

Insert **x** into the following checkboxes to confirm (eg. [x]):
- [ ] Bug fix.
- [ ] New feature.
- [ ] Enhancement.
- [ ] Unit test.
- [ ] Breaking change (a fix or feature that causes existing functionality to not work as expected).
- [ ] Requires documentation update.

## Testing

Please describe the tests you used to validate this pull request. Provide any relevant details for test configurations as well as any instructions to reproduce these results.

-

## Verification

Insert **x** into the following checkboxes to confirm (eg. [x]):
- [ ] I have self-reviewed my own code and conformed to the style guidelines of this project.
- [ ] New and existing tests pass locally with my changes.
- [ ] I have added tests for my fix or feature.
- [ ] I have made appropriate changes to the corresponding documentation.
- [ ] My code generates no new warnings.
- [ ] Any dependent changes have been made.

### CODE REVIEW

File: modApiServer/src/org/aion/api/server/rpc/ApiWeb3Aion.java [hunk 1]
```diff
@@ -2421,7 +2421,7 @@ public RpcMsg ops_getTransactionReceiptListByBlockHash(Object _params) {
             b = blockCache.get(new ByteArrayWrapper(blockHash));
         } catch (Exception e) {
             // Catch errors if send an incorrect tx hash
-            return new RpcMsg(null, RpcError.INVALID_REQUEST, "Invalid Request" + e.getMessage());
+            return new RpcMsg(null, RpcError.INVALID_REQUEST, "Invalid Request" + e);
```
Comment (AlexandraRoatis) [inline modApiServer/src/org/aion/api/server/rpc/ApiWeb3Aion.java]: let's update this to `"Invalid Request " + e +" " + e.getMessage()!=null? e.getMessage():''`

Comment (AionJayT) [reply modApiServer/src/org/aion/api/server/rpc/ApiWeb3Aion.java]: updated.

File: modApiServer/src/org/aion/api/server/rpc/ApiWeb3Aion.java [hunk 2]
```diff
@@ -2407,7 +2422,10 @@ public RpcMsg ops_getTransactionReceiptListByBlockHash(Object _params) {
             b = blockCache.get(new ByteArrayWrapper(blockHash));
         } catch (Exception e) {
             // Catch errors if send an incorrect tx hash
-            return new RpcMsg(null, RpcError.INVALID_REQUEST, "Invalid Request" + e.getMessage());
+            return new RpcMsg(
+                    null,
+                    RpcError.INVALID_REQUEST,
+                    "Invalid Request " + e + " " + e.getMessage() != null ? e.getMessage() : "");
```
Comment (aion-kelvin) [inline modApiServer/src/org/aion/api/server/rpc/ApiWeb3Aion.java]: I think putting `e` already includes the message that is in `e.getMessage()`

File: modApiServer/src/org/aion/api/server/pb/ApiAion0.java [hunk 3]
```diff
@@ -2448,9 +2419,7 @@ public ApiAion0(IAionChain ac) {
                             return ApiUtil.combineRetMsg(retHeader, rsp.toByteArray());
                         }
                     } catch (Exception e) {
-                        LOG.error(
-                                "ApiAion0.process.getBlocksByLatest exception: [{}]",
-                                e.getMessage());
+                        LOG.error("ApiAion0.process.getBlocksByLatest exception: [{}]", e);
```
Comment (aion-kelvin) [inline modApiServer/src/org/aion/api/server/pb/ApiAion0.java]: Since you the second arg is Exception, I think {} is not needed

**Pull request merged**